utils/oscap-podman

```diff
@@ -65,17 +65,53 @@ if grep -q "\-\-remediate" <<< "$@"; then
     die
 fi
 
+# When searching the scan target in a local container registry, we use internal
+# representation of container images in the following format:
+#     ID:REPOSITORY:TAG:
+# This is to prevent as many collisions as possible, for example, in case there
+# would be 'ubi8' and 'ubi8-minimal' images in the local container registry, it
+# is possible to provide just 'ubi8' as a scan target. We would be looking for
+# the 'ubi8:' and therefore 'ubi8-minimal' would not be matched.
+# This approach prevents most collisions, but there still might be some. For
+# example, having container images '1fedora' and '2fedora' and specifying
```

Comment:
I think we should support only exact matches. Why do we have support for substrings?

Reply:
Exact matches will not work in this case, only in case you would use the hash ID of the image (which is unique). For example, if you would have images like `registry.access.redhat.com/ubi8` and `registry.access.redhat.com/ubi8-minimal` and you would like to scan `ubi8` (not minimal), it would not help if you would type `registry.access.redhat.com/ubi8` because it would match also the `ubi8-minimal` so `oscap-podman` would need to produce an error. I find this annoying and the change in this commit tries to make it work in such cases.

On the other hand, I have no problem to change it so the error is printed, but IMO this is more user friendly.

Reply:
Another reason why exact matches can be a problem is that user can also specify a tag, for example, `ubi8:latest`, `ubi8:8.2.0`, etc.

Reply:
@jan-cerny Actually, I don't like my code, I will make it simpler and I would like to use `podman-image-exists`/`podman-container-exists`.

Reply:
I think that in your situation when you have images registry.access.redhat.com/ubi8 and registry.access.redhat.com/ubi8-minimal  and you request scan of registry.access.redhat.com/ubi8 the tool should scan only registry.access.redhat.com/ubi8 and not registry.access.redhat.com/ubi8-minimal. I understand registry.access.redhat.com/ubi8 as an unambiguous identifier, unless there are more of them with different tags.

If you want to add support for substring matches, then introduce some sort of glob or regular expressions. My opinion is that we don't need the substring matches, but we can have it if it works and is documented.

Let me know once you're done.